sub-packages/bionemo-llm/src/bionemo/llm/utils/callbacks.py

```diff
@@ -0,0 +1,100 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import os
+from typing import Any, Literal, Sequence
+
+import lightning.pytorch as pl
+import torch
+from lightning.pytorch.callbacks import BasePredictionWriter
+
+from bionemo.llm.lightning import batch_collator
+
+
+IntervalT = Literal["epoch", "batch"]
+
+
+class PredictionWriter(BasePredictionWriter, pl.Callback):
+    """A callback that writes predictions to disk at specified intervals during training."""
+
+    def __init__(self, output_dir: str | os.PathLike, write_interval: IntervalT):
+        """Initializes the callback.
+
+        Args:
+            output_dir (str): The directory where predictions will be written.
+            write_interval (str): The interval at which predictions will be written. (batch, epoch)
```

Comment:
maybe just remove the type hints in these docstring, you already have them in the function def